src/vsphere_cpi/lib/cloud/vsphere/vcenter_client.rb

```diff
@@ -185,6 +191,10 @@ def find_vm_by_name(datacenter_mob, vm_name)
       end
     end
 
+    def find_vm_by_uuid(datacenter_mob, vm_id)
+      @service_content.search_index.find_by_uuid(datacenter_mob, vm_id, true, true)
```

Comment:
Finds a virtual machine or host by BIOS or instance UUID.

vmSearch : If true, search for virtual machines, otherwise search for hosts.
instanceUuid : Should only be set when vmSearch is true. If specified, search for virtual machines whose instance UUID matches the given uuid. Otherwise, search for virtual machines whose BIOS UUID matches the given uuid.